agent/background.go

```diff
@@ -5,12 +5,14 @@ import (
 
 	"github.com/evergreen-ci/evergreen"
 	"github.com/evergreen-ci/evergreen/rest/client"
+	"github.com/evergreen-ci/evergreen/util"
 	"github.com/mongodb/grip"
 	"github.com/pkg/errors"
 	"golang.org/x/net/context"
 )
 
 func (a *Agent) startHeartbeat(ctx context.Context, tc *taskContext, heartbeat chan<- string) {
+	defer util.RecoverLogStackTraceAndContinue("heartbeat background process")
```

Comment:
If we do this on agents, will we ever know we panicked?

Reply:
It should show up the the log, but it does mean that the heartbeater will die, so maybe it's better to exit hard in these cases?

Reply:
I'm not sure. I think really we would want to be alerted if this happens.

Reply:
I think better to exit, because it'll mean the taskrunner will re-deploy the agent? if we log and continue, then we won't heartbeat until the next task starts, and may end up getting lots of File Busy errors.

Reply:
sgtm

Reply:
I changed my mind on this... I think its probably better to continue.

I'm going to merge and add a revision soon.